### ISSUE TITLE
Fake timers with long delay fail

### ISSUE THREAD

**Issue by leepowelldev** [kettanaito/msw#1505]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Node.js version

19.1.0

### Reproduction repository

https://github.com/leepowelldev/msw-delay-bug

### Reproduction steps

`npm test`

### Current behavior

Currently, when using fake timers in jest, adding a long delay will result in the response not being resolved. Very short delays, seem to work, but I think this is more by luck.

The repo includes three tests, no delay, short delay and a long delay... the long delay fails. I believe this is down to this line https://github.com/mswjs/msw/blob/7380011de47460212065ca41be1b5a1c60ed1b98/src/node/SetupServerApi.ts#L91 using a real timer for the `setTimeout`. This causes jest's fake timers and the real timer to get out of sync. 

Changing that line to use the faked global timeout (just for the delay!) allows us to advance any timers and have the promise resolve as expected.

```
if (response.delay) {
  await new Promise((resolve) => {
    globalThis.setTimeout(resolve, response.delay)
  })
}
```
Obviously I don't have deep knoledge of this library (thats for BTW!) so am a bit uncertain of the concequenses of making this change. It may mean users of fake timers need to advance the timers when using a delay - but I would sort of expect this behaviour as it's consistent with other timers. 

Happy to create a PR if needed.

### Expected behavior

All tests to pass regardless of the delay duration.

**Comment by leepowellnbs** [comment]
Update - I did figure out a workaround is to use real timers to wait for the delay to complete... not ideal, but seems to work around the issue.
```
function wait(ms: number) {
  return new Promise((resolve) => {
    jest.requireActual('timers').setTimeout(resolve, ms);
  });
}
```
Then in my test, wait for the delay duration to complete with real timers.
```
await act(() => wait(1000));
```

**Comment by kettanaito** [comment]
The issue doesn't seem to be reproducible on the latest version of MSW:

```
 PASS  src/App.test.tsx
  ✓ data is fetched with no delay (170 ms)
  ✓ data is fetched with short delay (35 ms)
  ✓ data is fetched with long delay (33 ms)

Test Suites: 1 passed, 1 total
Tests:       3 passed, 3 total
Snapshots:   0 total
Time:        2.046 s
```

Please [upgrade](https://mswjs.io/docs/migrations/1.x-to-2.x) and use [`jest-fixed-jsdom`](https://github.com/mswjs/jest-fixed-jsdom) to fix JSDOM to work correctly. 

Here's the diff for your repo:

```diff
jest.config.js
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

────────────────────┐
8: export default { │
────────────────────┘
  fakeTimers: {
    enableGlobally: true,
  },
-  testEnvironment: 'jsdom',
+  testEnvironment: 'jest-fixed-jsdom',
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
};

package.json
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

────┐
11: │
────┘
  },
  "dependencies": {
    "cross-fetch": "^3.1.5",
-    "msw": "^0.49.2",
+    "jest-fixed-jsdom": "^0.0.9",
+    "msw": "^2.6.8",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },

src/App.test.tsx
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

───┐
1: │
───┘
import React from 'react';
import { App } from './App';
import { screen, render, waitFor } from '@testing-library/react';
-import { rest } from 'msw';
+import { delay, http } from 'msw';
import { setupServer } from 'msw/node';

const server = setupServer();

───────────────────────┐
20: beforeEach(() => { │
───────────────────────┘

test('data is fetched with no delay', async () => {
  server.use(
-    rest.get('https://mydomain.com', (_, res, ctx) => {
-      return res(
-        ctx.json({
-          message: 'success',
-        })
-      );
+    http.get('https://mydomain.com', () => {
+      return Response.json({
+        message: 'success',
+      });
+    })
  );


────────────────────────────────────────────────────────┐
40: test('data is fetched with no delay', async () => { │
────────────────────────────────────────────────────────┘

test('data is fetched with short delay', async () => {
  server.use(
-    rest.get('https://mydomain.com', (_, res, ctx) => {
-      return res(
-        ctx.delay(),
-        ctx.json({
-          message: 'success',
-        })
-      );
+    http.get('https://mydomain.com', async () => {
+      await delay();
+      return Response.json({
+        message: 'success',
+      });
+    })
  );


───────────────────────────────────────────────────────────┐
61: test('data is fetched with short delay', async () => { │
───────────────────────────────────────────────────────────┘

test('data is fetched with long delay', async () => {
  server.use(
-    rest.get('https://mydomain.com', (_, res, ctx) => {
-      return res(
-        ctx.delay(1000),
-        ctx.json({
-          message: 'success',
-        })
-      );
+    http.get('https://mydomain.com', async () => {
+      await delay(1000);
+      return Response.json({
+        message: 'success',
+      });
+    })
  );
```
